crates/update-notifier/src/ui.rs

```diff
@@ -0,0 +1,132 @@
+use colored::*;
+use terminal_size::{terminal_size, Width};
+
+mod utils;
+
+const PADDING: usize = 8;
+const TOP_LEFT: &str = "╭";
+const TOP_RIGHT: &str = "╮";
+const BOTTOM_LEFT: &str = "╰";
+const BOTTOM_RIGHT: &str = "╯";
+const HORIZONTAL: &str = "─";
+const VERTICAL: &str = "│";
+const SPACE: &str = " ";
+
+enum BorderAlignment {
+    Divider,
+    Top,
+    Bottom,
+}
+
+fn x_border(width: usize, position: BorderAlignment) {
+    match position {
+        BorderAlignment::Top => {
+            println!(
+                "{}{}{}",
+                TOP_LEFT.yellow(),
+                HORIZONTAL.repeat(width).yellow(),
+                TOP_RIGHT.yellow()
+            );
+        }
+        BorderAlignment::Bottom => {
+            println!(
+                "{}{}{}",
+                BOTTOM_LEFT.yellow(),
+                HORIZONTAL.repeat(width).yellow(),
+                BOTTOM_RIGHT.yellow()
+            );
+        }
+        BorderAlignment::Divider => {
+            println!("{}", HORIZONTAL.repeat(width).yellow(),);
+        }
+    }
+}
+
+pub fn rectangle(text: &str) {
+    let size = terminal_size();
+    let lines: Vec<&str> = text.split("\n").map(|line| line.trim()).collect();
+    // get the display width of each line so we can center it within the box later
+    let lines_display_width: Vec<usize> = lines
+        .iter()
+        .map(|line| utils::get_display_length(line).unwrap())
+        .collect();
+    let longest_line = lines_display_width.iter().max().unwrap().to_owned();
```

Comment:
In general if a type implements `Copy` you should use that functionality over `Clone`/`ToOwned` to indicate it's a cheap operation.

```suggestion
    let longest_line = lines_display_width.iter().max().copied().unwrap_or_default();
```

Reply:
👍🏼 sounds good, thanks!